src/lib/components/Navigation/Navigation.stories.tsx

```diff
@@ -0,0 +1,83 @@
+import { Meta } from "@storybook/react";
+
+import { Navigation } from "./Navigation";
+
+const meta: Meta<typeof Navigation> = {
+  title: "Components/Navigation",
+  component: Navigation,
+  render: () => (
+    <Navigation>
+      <Navigation.Bar>
```

Comment:
Might be worth exploring separation of the Navigation.Bar and Navigation.Nav components. This nesting seems unnecessary. For example:

```
<NavigationBar>
  <NavigationBar.Header>
    ...
  </NavigationBar.Header>
</NavigationBar>

<Navigation isCollapsed={false}>
  <Navigation.Drawer>
    ...
  </Navigation.Drawer>
</Navigation>
```